keps/sig-windows/1981-windows-privileged-container-support/README.md

```diff
@@ -459,87 +464,136 @@ Windows privileged containers will be implemented with [Job Objects](https://doc
+On Windows, all containers in a pod must be privileged. Because of this behavior and because `WindowsSecurityContextOptions` already exists on both `PodSecurityContext` and `Container.SecurityContext` Windows containers will use this new field instead of solely relying on the existing `privileged` field which only exists on `SecurityContext`. Because many policy tools currently look for the existing `SecurityContext.privileged` field we will add validation to requrire this field also be set on all containers in a privileged pod.
 
-https://github.com/kubernetes/kubernetes/blob/d20fd4088476ec39c5ae2151b8fffaf0f4834418/pkg/kubelet/container/helpers.go#L227-L229
+API validation will ensure that if `privileged` is set in `Container.SecurityContext.WindowsSecurityContextOptions` for any container then then `privileged` must also set in `PodSecurityContext.WindowsSecurityContextOptions`. Current behavior applies pod `WindowsSecurityContextOptions` to all container level `WindowsSecurityContextOptions` if not specified.
 
-https://github.com/kubernetes/kubernetes/blob/a9f1d72e1de6450b890a0c0e451725468f54f515/pkg/kubelet/kuberuntime/kuberuntime_sandbox.go#L98 
+#### Host Network Mode
 
-In the case of windows this field would need to be addressed.  During the alpha stage the proposal for this field would be to add documentation that note the field is only applicable to Linux.  In addition the check for hostnetwork mode will need to be updated to check the privileged flag on windows PodSpec.  This will be implemented in the following way, with validation loosened in the future if required:
- - this pod must run on a windows host, and kubelets must reject it if not on windows hosts
- - all pods marked privileged on windows must have host network enabled, if not the pod does not validate
+Host Network mode for privileged Windows containers will always be enabled, as the pod will automatically get the host IP.
 
-In beta there is a possibility to enable the privileged container to be a part of a different network component.  If this feature is enabled we will use the existing Pod HostNetwork field to enable/disable.
+Privileged Windows containers will be unable to align to pod namespaces due to limitations in the Windows OS. This functionality will likely be enabled in the future through a new KEP.
 
-#### <b> ContainerD Support Only</b>
-There are no plans to update Docker to have support for Privileged containers due to requirements for support HCSv2.  The additional fields for docker would be ignored. The default value should be false.
+Because of this we will require that `hostNetwork` is set to `true` when scheduling privileged pods. This will allow existing policy tools to detect and act on privileged Windows containers without any updates. In the future if/when functionality is added to support joining privileged containers to pod networks this validation will be revisited.
```

Comment:
"scheduling time" meaning the validation will be on the apiserver side? Will there be validation on the kubelet side? Just want to know if pods from other sources (e.g., file and http) will be validated, and if not, how they would fail.

Reply:
We could add a check in either the kubelet or containerd to enforce this. Let me look into which spot would be better.
Thanks for pointing this out!

Reply:
Updated the KEP to add some kubelet checks for Windows only for additional validation.